File: java/money/ch07/src/test/java/com/thesoftwaregorilla/tdd/money/FrancTest.java
package com.thesoftwaregorilla.tdd.money;

import org.junit.jupiter.api.Test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNotEquals;

public class FrancTest {

    @Test
    public void testConstruction() {
        Franc five = new Franc(5);
        assertNotNull(five);
        assertEquals(5, five.getAmount());
        Franc ten = new Franc(10);
        assertNotNull(ten);
        assertEquals(10, ten.getAmount());
    }

    @Test
    public void testMultiplication() {
        Franc five = new Franc(5);
        assertEquals(new Franc(10), five.times(2));
        assertEquals(new Franc(15), five.times(3));
    }

}

File: css/ch10/Money.Tests/FrancTest.cs
using NUnit.Framework;
using TheSoftwareGorilla.TDD.Money;

namespace TheSoftwareGorilla.TDD.Money.Tests;

public class FrancTests
{
    [SetUp]
    public void Setup()
    {
    }


    [Test]
    public void TestMultiplication()
    {
        Money five = Money.Franc(5);
        Assert.That(Money.Franc(10), Is.EqualTo(five.Times(2)));
        Assert.That(Money.Franc(15), Is.EqualTo(five.Times(3)));
    }

}